commands/builds/create.js

```diff
@@ -43,25 +43,32 @@ function compressSource (context, dir, tempFile) {
   })
 }
 
-async function uploadDirToSource (context, heroku, dir) {
-  let tempFilePath = path.join(os.tmpdir(), uuid.v4() + '.tar.gz')
+async function uploadDirToSource (context, heroku, dir, filePath) {
```

Comment:
We will always have either a filePath or a dir, never both, which are two quite similar arguments. It looks like we could get rid of one.
Possibly by renaming the argument to `path`, and detecting if it's a directory or not and changing the behavior?

Then, the method would also be renamed to `uploadToSource`, as it doesn't only upload a directory.